python/cuxfilter/charts/core/aggregate/core_number_chart.py

```diff
@@ -0,0 +1,297 @@
+import numpy as np
+from ..core_chart import BaseChart
+from ....layouts import chart_view
+
+
+class BaseNumberChart(BaseChart):
+    stride = 1
+
+    @property
+    def use_data_tiles(self):
+        return self.expression is None
+
+    @property
+    def is_datasize_indicator(self):
+        return not (self.x or self.expression)
+
+    @property
+    def name(self):
+        value = (self.x or self.expression) or ""
+        chart_type = self.chart_type or ""
+
+        return value + "_" + chart_type
```

Comment:
In case `chart_type` ends up the empty string this will return "foo_" with a trailing underscore

Is that intended? (Is it intended that `self.chart_type` be none?)

Reply:
Actually no, `chart_type` is not intended to be None. That's why I set it as either `number_chart` or `number_chart_widget` in the [constructor](https://github.com/rapidsai/cuxfilter/pull/228/files/d748e138266f24e5a4a01007de77295fea4d4736#diff-52f1ec033c754dc4500c5435b1c6cdb55769fca1c77b09b52add4fd86a5c05cbR55).

Reply:
Its not intended to be None. The [constructor](https://github.com/rapidsai/cuxfilter/pull/228/files/d748e138266f24e5a4a01007de77295fea4d4736#diff-52f1ec033c754dc4500c5435b1c6cdb55769fca1c77b09b52add4fd86a5c05cbR55) sets the chart_type to either `number_chart` or `number_chart_widget`. 

Ah, I get it, the line  `chart_type = self.chart_type or ""` doesn't make sense in this scenario then